### ISSUE TITLE
Validation error when using load balancing algorithm with arguments

### ISSUE THREAD

**Issue by fabrichter** [haproxytech/kubernetes-ingress#104]
Hi, 
it seems like parsing load balancing methods with arguments fails.

Steps to reproduce:

```
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: web-ingress
  namespace: default
  annotations:
    haproxy.org/load-balance: "url_param sessionId"
```

Error message:
```
2019/10/25 11:24:16 controller-annotations.go:97 balance algorithm: validation failure list:
algorithm in body should be one of [roundrobin static-rr leastconn first source uri url_param random]
```

**Comment by oktalz** [comment]
hi @fabrichter 
it appears this is a bug, thx for reporting this.

**Comment by devGenie** [comment]
Hi, this bug is still occuring. Has there been any update on this issue? Thanks

**Comment by lpranam** [comment]
any workaround for this I am still facing this issue! also the same issue with `hdr(header)`

**Comment by lpranam** [comment]
a simple workaround


```
annotations:
  haproxy.org/check: "true"
  haproxy.org/forwarded-for: "true"
  backend-config-snippet: |
    balance url_param param check_post
```

**Comment by chasmosis** [comment]
fought this for 1/2 the day today before finding this issue.  @lpranam 's workaround fixed me up for load-balance on x-forwarded-for header.  Thanks @lpranam

**Comment by ivanmatmati** [comment]
Hi, this bug is fixed since 1.7.
